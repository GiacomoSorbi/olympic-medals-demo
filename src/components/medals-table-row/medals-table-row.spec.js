import * as React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import MedalsTableRow from './'

describe('MedalsTableRow should', () => {
  it('exist', () => {
    expect(MedalsTableRow).not.toBeUndefined()
  })

  it('have a consistent structure', () => {
    const wrapper = mount(
      <MedalsTableRow
        countryData={{
          name: 'PippiLand',
          gold: 99,
          silver: 66,
          bronze: 33,
          total: 198,
        }}
        onEdit={jest.fn()}
      />,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
