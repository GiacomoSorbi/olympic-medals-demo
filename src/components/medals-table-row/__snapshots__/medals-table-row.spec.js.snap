// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedalsTableRow should have a consistent structure 1`] = `
<MedalsTableRow
  countryData={
    Object {
      "bronze": 33,
      "gold": 99,
      "name": "PippiLand",
      "silver": 66,
      "total": 198,
    }
  }
  onEdit={[MockFunction]}
>
  <tr>
    <th
      className="text-left align-middle"
      scope="row"
    >
       
      PippiLand
    </th>
    <td
      className="text-right border align-middle"
    >
      99
    </td>
    <td
      className="text-right border align-middle"
    >
      66
    </td>
    <td
      className="text-right border align-middle"
    >
      33
    </td>
    <td
      className="text-right border  align-middle"
    >
      198
    </td>
    <td
      className="align-middle border"
    >
      <Button
        color="primary"
        data-country="PippiLand"
        onClick={[MockFunction]}
        tag="button"
      >
        <button
          aria-label={null}
          className="btn btn-primary"
          data-country="PippiLand"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="edit row"
            className="edit-icon"
            src="edit.svg"
          />
        </button>
      </Button>
    </td>
  </tr>
</MedalsTableRow>
`;
